### PR TITLE
fix: Validate resource file also for data elements [DHIS2-9208]

### PR DESCRIPTION
**Pull request by enricocolasante**

### CODE REVIEW

File: dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/preheat/supplier/FileResourceSupplier.java [hunk 1]
```diff
@@ -87,4 +99,15 @@ private void collectResourceIds( List<String> fileResourceAttributes, List<Strin
...
     }
+
+    private void collectResourceIds( List<String> fileResourceDataElements, List<String> fileResourceIds,
+        Set<DataValue> dataElements )
+    {
+        dataElements.forEach( de -> {
+            if ( fileResourceDataElements.contains( de.getDataElement() ) && !StringUtils.isEmpty( de.getValue() ) )
+            {
+                fileResourceIds.add( de.getValue() );
+            }
```
Comment (lucaCambi77) [inline dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/preheat/supplier/FileResourceSupplier.java:110]: if we add these filter, we need to test it all  cases like empty value and mixed cases with existing resource

File: dhis-2/dhis-services/dhis-service-tracker/src/test/java/org/hisp/dhis/tracker/preheat/supplier/FileResourceSupplierTest.java [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+    @Before
+    public void setUp()
+    {
+        DataElement numericDataElement = createDataElement( 'A' );
+        numericDataElement.setUid( "numericDataElement" );
+        numericDataElement.setValueType( ValueType.NUMBER );
+
+        DataElement fileResourceDataElement = createDataElement( 'B' );
+        fileResourceDataElement.setUid( "fileResourceDataElement" );
+        fileResourceDataElement.setValueType( ValueType.FILE_RESOURCE );
```
Comment (lucaCambi77) [inline dhis-2/dhis-services/dhis-service-tracker/src/test/java/org/hisp/dhis/tracker/preheat/supplier/FileResourceSupplierTest.java:79]: we can extract static string for uid so it is more clear

**Review by lucaCambi77 — CHANGES_REQUESTED** [review]

**Review by gnespolino — APPROVED** [review]

**Review by ameenhere — APPROVED** [review]

File: dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/preheat/supplier/FileResourceSupplier.java (hunk 1 above)
Comment (enricocolasante) [reply dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/preheat/supplier/FileResourceSupplier.java:110]: Done

File: dhis-2/dhis-services/dhis-service-tracker/src/test/java/org/hisp/dhis/tracker/preheat/supplier/FileResourceSupplierTest.java (hunk 2 above)
Comment (enricocolasante) [reply dhis-2/dhis-services/dhis-service-tracker/src/test/java/org/hisp/dhis/tracker/preheat/supplier/FileResourceSupplierTest.java]: Done

**Review by lucaCambi77 — APPROVED** [review]
